cheddar/cheddar-integration-aws/src/main/java/com/clicktravel/infrastructure/messaging/aws/sqs/SqsBasicMessageQueue.java

```diff
@@ -36,4 +50,48 @@ protected BasicMessage toMessage(final Message sqsMessage) {
         return new SimpleBasicMessage(sqsMessage.getBody(), sqsMessage.getMessageId(), sqsMessage.getReceiptHandle());
     }
 
+    @Override
+    public List<BasicMessage> receive() throws MessageReceiveException {
+        final List<BasicMessage> receivedMessages = super.receive();
+        logReceivedMessages(receivedMessages);
+        return receivedMessages;
+    }
+
+    @Override
+    public List<BasicMessage> receive(final int waitTimeSeconds, final int maxMessages) throws MessageReceiveException {
+        final List<BasicMessage> receivedMessages = super.receive(waitTimeSeconds, maxMessages);
+        logReceivedMessages(receivedMessages);
+        return receivedMessages;
+    }
+
+    @Override
+    public void send(final BasicMessage message) throws MessageSendException {
+        logSendMessage(message);
+        super.send(message);
+    }
+
+    @Override
+    public void sendDelayedMessage(final BasicMessage message, final int delaySeconds) throws MessageSendException {
+        logSendMessage(message);
+        super.sendDelayedMessage(message, delaySeconds);
+    }
+
+    private void logReceivedMessages(final List<BasicMessage> receivedMessages) {
+        if (isLogReceivedMessages()) {
+            for (final BasicMessage message : receivedMessages) {
+                try {
+                    LOGGER.debug("MSG-RECV [{}] [{}]", message.getClass().getSimpleName(), message.getBody());
```

Comment:
`BasicMessage`s have no type property, so just use empty string for logging the message type (the second parameter in this method call)